include/hipSYCL/sycl/libkernel/sscp/group_functions.hpp

```diff
@@ -67,15 +69,14 @@ __acpp_group_barrier(sub_group g,
 
 // broadcast
 
-template <int Dim, typename T>
+template <int Dim, typename T, std::enable_if_t<sizeof(T) <= 8, int> = 0 >
 HIPSYCL_BUILTIN 
-T __acpp_group_broadcast(
+ std::enable_if_t<(sizeof(T) <= 8), T> __acpp_group_broadcast(
```

Comment:
This was intended as a functional change. (At least one of `enable_if_t`s :P) Without this, in case T is `sycl::vec<int, 3>` for instance the template will be ambiguous with the specialization meant for vectors. 

Removing this change yields:
```
/data/sbalint/acpp_gmx_practical/AdaptiveCpp-upstreaming/build/install/bin/../include/AdaptiveCpp/hipSYCL/sycl/libkernel/group_functions.hpp:62:3: error: call to '__acpp_group_broadcast' is ambiguous
  HIPSYCL_RETURN_DISPATCH_GROUP_ALGORITHM(__acpp_group_broadcast, g, x,
.....

/data/sbalint/acpp_gmx_practical/AdaptiveCpp-upstreaming/build/install/bin/../include/AdaptiveCpp/CL/../hipSYCL/sycl/libkernel/sscp/group_functions.hpp:74:4: note: candidate function [with Dim = 1, T = hipsycl::sycl::vec<int, 3>]
 T __acpp_group_broadcast(
   ^
/data/sbalint/acpp_gmx_practical/AdaptiveCpp-upstreaming/build/install/bin/../include/AdaptiveCpp/CL/../hipSYCL/sycl/libkernel/sscp/group_functions.hpp:137:1: note: candidate function [with T = int, N = 3, Group = hipsycl::sycl::group<1>]
__acpp_group_broadcast(
```